File: testdata/jennies/rawtypes/field_with_struct_with_defaults/PHPRawTypes/src/Defaults/NestedStruct.php
<?php

namespace Grafana\Foundation\Defaults;

class NestedStruct implements \JsonSerializable
{
    public string $stringVal;

    public int $intVal;

    /**
     * @param string|null $stringVal
     * @param int|null $intVal
     */
    public function __construct(?string $stringVal = null, ?int $intVal = null)
    {
        $this->stringVal = $stringVal ?: "";
        $this->intVal = $intVal ?: 0;
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->stringVal !== $other->stringVal) {
            return false;
        }
    
        if ($this->intVal !== $other->intVal) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{stringVal?: string, intVal?: int} $inputData */
        $data = $inputData;
        return new self(
            stringVal: $data["stringVal"] ?? null,
            intVal: $data["intVal"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->stringVal = $this->stringVal;
        $data->intVal = $this->intVal;
        return $data;
    }
}

File: testdata/jennies/rawtypes/field_with_struct_with_defaults/JavaRawTypes/defaults/NestedStruct.java
package defaults;

import java.util.Objects;

public class NestedStruct {
    public String stringVal;
    public Long intVal;
    public NestedStruct() {
        this.stringVal = "";
        this.intVal = 0L;
    }
    public NestedStruct(String stringVal,Long intVal) {
        this.stringVal = stringVal;
        this.intVal = intVal;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof NestedStruct)) return false;
        NestedStruct o = (NestedStruct) other;
        if (!Objects.equals(this.stringVal, o.stringVal)) return false;
        if (!Objects.equals(this.intVal, o.intVal)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.stringVal, this.intVal);
    }
}
